rp_yal/tests/test_utils.py

```diff
@@ -495,3 +496,38 @@ def test_get_first_matching_content_set_no_matches(self):
         contentset_id = utils.get_first_matching_content_set(TEST_CONTENT_SETS, fields)
 
         self.assertEqual(contentset_id, 3)
+
+
+class GetUniquePageSeenIds(TestCase):
+    def setUp(self):
+        self.org = create_org()
+
+    def tearDown(self):
+        redis_conn.delete("page_seen_ids_27831231234")
+
+    @responses.activate
+    def test_get_unique_page_seen_ids_cache(self):
+        msisdn = "27831231234"
+        responses.add(
+            method=responses.GET,
+            url="http://contentrepo/api/v2/custom/pageviews/",
+            json={
+                "results": [
+                    {"page": 164},
+                    {"page": 165},
+                    {"page": 166},
+                ],
+            },
+            status=200,
+            match=[
+                responses.matchers.query_param_matcher(
+                    {"data__user_addr": msisdn, "unique_pages": "true"}
+                )
+            ],
+        )
+
+        ids = utils.get_unique_page_seen_ids(self.org, msisdn)
+        self.assertEqual(ids, [164, 165, 166])
+        ids = utils.get_unique_page_seen_ids(self.org, msisdn)
```

Comment:
We are repeating the assert here

Reply:
Yes, calling twice to make sure the second time round the values comes from the cache and it's not doing another call to contentrepo and the results are still the same